fbw-a32nx/src/systems/fmgc/src/flightplanning/plans/BaseFlightPlan.ts

```diff
@@ -908,14 +909,31 @@ export abstract class BaseFlightPlan<P extends FlightPlanPerformanceData = Fligh
           if (previousElement.isXI()) {
             prevSegment.allLegs.splice(prevIndexInSegment, 1);
           }
-        } else if (nextElement.isDiscontinuity === false) {
+        } else if (nextElement?.isDiscontinuity === false) {
           if (previousElement.terminatesWithWaypoint(nextElement.terminationWaypoint())) {
             // Disco with same point before and after it
             this.mergeConstraints(previousElement, nextElement);
 
             this.removeRange(index, index + 2);
           } else {
             // Regular disco
+            if (nextElement.isXF()) {
+              const [nextSegment, nextIndexInSegment] = this.segmentPositionForIndex(index + 1);
+
+              // Convert next element to TF
+              // We do this because we get the wrong turn direction sometimes if we keep it a CF leg for example
+              const newNextElement = FlightPlanLeg.fromEnrouteFix(
+                nextSegment,
+                nextElement.terminationWaypoint(),
+                nextElement.annotation,
+              )
+                .withDefinitionFrom(nextElement)
+                .withPilotEnteredDataFrom(nextElement);
+
+              nextSegment.allLegs.splice(nextIndexInSegment, 1, newNextElement);
```

Comment:
If we modify segments other than the one the leg being removed is in, we should add a sync call for that segment at the end of this function.